metals/src/main/scala/scala/meta/internal/metals/RamboSymbolSearch.scala

```diff
@@ -0,0 +1,86 @@
+package scala.meta.internal.metals
+
+import scala.meta.pc.SymbolSearch
+import scala.meta.pc.SymbolDocumentation
+import java.{util => ju}
+import org.eclipse.lsp4j.Location
+import scala.meta.pc.SymbolSearchVisitor
+import scala.meta.pc.SymbolSearch.Result
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import scala.meta.internal.mtags.OnDemandSymbolIndex
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.Symbol
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.mtags.Mtags
+
+class RamboSymbolSearch(workspace: AbsolutePath) extends SymbolSearch {
+
+  private val dependencySourceCache =
+    new TrieMap[AbsolutePath, ju.List[String]]()
+
+  private val scalaVersion = BuildInfo.scala212
+
+  private val fetch = Fetch
+    .create()
+    .withMainArtifacts()
+    .withClassifiers(Set("sources", "_").asJava)
+    .withDependencies(
+      Dependency.of("org.scala-lang", "scala-library", scalaVersion)
+    )
+
+  private val jars = fetch
+    .fetch()
+    .asScala
+    .map(file => AbsolutePath(file.toPath))
+
+  private val (sources, classpath) =
+    jars.partition(_.toString.endsWith("-sources.jar"))
+
+  private val scalaDependency =
+    ClasspathSearch.fromClasspath(classpath.toList.map(_.toNIO))
+
+  private val index = OnDemandSymbolIndex()
+  sources.foreach(index.addSourceJar)
+
+  private val docs = new Docstrings(index)
+
+  def documentation(symbol: String): ju.Optional[SymbolDocumentation] =
+    docs.documentation(symbol)
+
+  def definition(x: String): ju.List[Location] = {
+    index
+      .definition(Symbol(x))
+      .map { definition =>
+        val destinationPath = definition.path.toFileOnDisk(workspace)
```

Comment:
This logic looks duplicated from some other location, could we share the code somehow?